front/lib/registry.ts

```diff
@@ -1,20 +1,21 @@
-import { DustAppType } from "../../../front/lib/dust_api";
+import type { DustAppType } from "@dust-tt/types";
 
-const PRODUCTION_DUST_APPS_WORKSPACE_ID = "78bda07b39";
+import config, { PRODUCTION_DUST_WORKSPACE_ID } from "@app/lib/api/config";
+
+const DUST_APPS_WORKSPACE_ID = config.getDustAppsWorkspaceId();
+
+export const isProductionDustAppsWorkspaceId = () =>
+  PRODUCTION_DUST_WORKSPACE_ID === DUST_APPS_WORKSPACE_ID;
 
 export type Action = {
   app: DustAppType;
   config: { [key: string]: unknown };
 };
 
-const createActionRegistry = <K extends string, R extends Record<K, Action>>(
-  registry: R
-) => registry;
-
-export const DustProdActionRegistry = createActionRegistry({
+export const DustProdActionRegistry = {
   "assistant-v2-multi-actions-agent": {
     app: {
-      workspaceId: PRODUCTION_DUST_APPS_WORKSPACE_ID,
+      workspaceId: DUST_APPS_WORKSPACE_ID,
```

Comment:
What does PRODUCTION_DUST_APPS_WORKSPACE_ID become? Why change the name?

isProductionDustAppsWorkspaceId is also almost impossible to understand.

I think we need to step back a bit and adopt a slightly cleaner design:
- I would keep this untouched as this is the real real real definition of our dust-apps.
- I would instead move the logic of overriding the workspace id in runAction/runActionStreamed with a very very clear if (isDevelopment) encapsulating all workspace switching logic.
- I would rename DUST_APPS_WORKSPACE_ID into `DEVELOPMENT_DUST_APPS_WORKSPACE_ID` which will make this variable much more understandable wrt to `PRODUCTION_DUST_APPS_WORKSPACE_ID`

Reply:
Thinking more about this do we really need some code here?

Isn't it just a matter of setting `PRODUCTION_DUST_APPS_WORKSPACE_ID` to your local workspace and setting `DUST_PROD_API` (to localhost) locally?

The only cumbersome thing here is that you have to set `PRODUCTION_DUST_APPS_WORKSPACE_ID` after creating your workspace and syncing but that's not that bad.

The real automation we want is that we create defaul dust-apps workspace when setting up the db in development and sync the apps to it no?

Reply:
PRODUCTION_DUST_APPS_WORKSPACE_ID is the workspace id of dust-apps in production - it can't be modified, it's a constant set to 78bda07b39 . It's the default value of DUST_APPS_WORKSPACE_ID, and also used in isProductionDustAppsWorkspaceId to check if we changed it or not.

Creating the dust-apps into a local workspace is done by a script, but we need to be able to use them without changing the code. In terms of flow, if you start form scratch, yes you have to create your workspace first, then sync the apps and set your DUST_APPS_WORKSPACE_ID .

Reply:
- I've changed the code so that the registry part stays unmodified. I can't do the change in runAction/runActionStreams as it's in types, so I instead filled the `createActionRegistry` method to replace the workspace id only if we are in development and if we set a development one.
- the env var to set is now DEVELOPMENT_DUST_APPS_WORKSPACE_ID
- I've removed the isProductionDustAppsWorkspaceId - just checking now in retrival/process if we have set a development workspace id.